src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,357 @@
+        = new List<KeyValuePair<string, Color[]>>();
+
+    private readonly List<TweakedColourPickerPreset> presets = new List<TweakedColourPickerPreset>();
+
+    private HSlider sliderROrH;
+    private HSlider sliderGOrS;
+    private HSlider sliderBOrV;
+    private HSlider sliderA;
+    private ToolButton pickerButton;
+    private CheckButton hsvCheckButton;
+    private CheckButton rawCheckButton;
+    private LineEdit htmlColourEdit;
+    private HSeparator separator;
+    private GridContainer presetsContainer;
+    private TextureButton addPresetButton;
+
+    private bool hsvButtonEnabled = true;
+    private bool rawButtonEnabled = true;
+    private bool presetsEnabled = true;
+    private bool presetsVisible = true;
+
+    /// <summary>
+    ///   Decide if user can toggle HSV CheckButton to switch HSV mode.
+    /// </summary>
+    [Export]
+    public bool HSVButtonEnabled
+    {
+        get => hsvButtonEnabled;
+        set
+        {
+            hsvButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    /// <summary>
+    ///   Decide if user can toggle Raw CheckButton to switch Raw mode.
+    /// </summary>
+    [Export]
+    public bool RawButtonEnabled
+    {
+        get => rawButtonEnabled;
+        set
+        {
+            rawButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool HsvMode
+    {
+        get => base.HsvMode;
+        set
+        {
+            if (value && RawMode)
+                return;
+
+            base.HsvMode = value;
+            UpdateButtonsState();
+
+            // Update HSV sliders' tooltips
+            UpdateTooltips();
+        }
+    }
+
+    [Export]
+    public new bool RawMode
+    {
+        get => base.RawMode;
+        set
+        {
+            if (value && HsvMode)
+                return;
+
+            base.RawMode = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool PresetsEnabled
+    {
+        get => presetsEnabled;
+        set
+        {
+            presetsEnabled = value;
+
+            if (addPresetButton == null)
+                return;
+
+            addPresetButton.Disabled = !value;
+        }
+    }
+
+    [Export]
+    public new bool PresetsVisible
+    {
+        get => presetsVisible;
+        set
+        {
+            presetsVisible = value;
+
+            if (presetsContainer == null)
+                return;
+
+            separator.Visible = value;
+            presetsContainer.Visible = value;
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        // Hide replaced native controls. Can't delete them because it will crash Godot.
+        GetChild(4).GetChild<Control>(4).Hide();
+        GetChild<Control>(5).Hide();
+        GetChild<Control>(6).Hide();
+        GetChild<Control>(7).Hide();
+
+        // Get controls
+        var baseControl = GetChild(4);
```

Comment:
This was already fetched on line 138, so this variable could be moved to line 136 to avoid that duplicate lookup.